command/genesis/polybft_params.go

```diff
@@ -0,0 +1,224 @@
+	if len(p.bootnodes) == 0 {
+		for i, validator := range validatorsInfo {
+			// /ip4/127.0.0.1/tcp/10001/p2p/16Uiu2HAm9r5oP8Dmfsqbp1w2LdPU4YSFggKvwEmT6aTpWU8c8R13
+			bnode := fmt.Sprintf("/ip4/%s/tcp/%d/p2p/%s", "127.0.0.1", bootnodePortStart+i, validator.NodeID)
+			chainConfig.Bootnodes = append(chainConfig.Bootnodes, bnode)
+		}
+	}
+
+	// Premine accounts
+	if err := fillPremineMap(chainConfig.Genesis.Alloc, p.premine); err != nil {
+		return nil, err
+	}
+
+	return chainConfig, nil
+}
+
+func (p *genesisParams) getGenesisValidators(validators []GenesisTarget) (result []*polybft.Validator) {
+	if len(p.validators) > 0 {
+		for _, validator := range p.validators {
+			parts := strings.Split(validator, ":")
+			if len(parts) != 2 || len(parts[0]) != 32 || len(parts[1]) < 2 {
+				continue
+			}
+
+			result = append(result, &polybft.Validator{
+				Address: types.Address(ethgo.HexToAddress(parts[0])),
+				BlsKey:  parts[1],
+			})
+		}
+	} else {
+		for _, validator := range validators {
+			pubKeyMarshalled := validator.Account.Bls.PublicKey().Marshal()
+
+			result = append(result, &polybft.Validator{
+				Address: types.Address(validator.Account.Ecdsa.Address()),
+				BlsKey:  hex.EncodeToString(pubKeyMarshalled),
+			})
+		}
+	}
+
+	return result
+}
+
+func deployContracts() ([]polybft.SmartContract, error) {
+	predefinedContracts := []struct {
+		name     string
+		input    []interface{}
+		expected ethgo.Address
+		chain    string
+	}{
+		{
+			// Validator smart contract
+			name:     "Validator",
+			expected: ValidatorSetAddr,
+			chain:    "child",
+		},
+		{
+			// Bridge in the sidechain
+			name:     "SidechainBridge",
+			expected: SidechainBridgeAddr,
+			chain:    "child",
+		},
+		{
+			// Target ERC20 token
+			name:     "MintERC20",
+			expected: sidechainERC20Addr,
+			chain:    "child",
+		},
+		{
+			// Bridge wrapper for ERC20 token
+			name: "ERC20Bridge",
+			input: []interface{}{
+				sidechainERC20Addr,
+			},
+			expected: SidechainERC20BridgeAddr,
+			chain:    "child",
+		},
+	}
+
+	result := make([]polybft.SmartContract, 0, len(predefinedContracts))
+
+	// to call the init in validator smart contract we do not need much more context in the evm object
+	// that is why many fields are set as default (as of now).
+	for _, contract := range predefinedContracts {
+		artifact, err := polybftcontracts.ReadArtifact(contract.chain, contract.name)
+		if err != nil {
+			return nil, err
+		}
+
+		input, err := artifact.DeployInput(contract.input)
+		if err != nil {
+			return nil, err
+		}
+
+		smartContract := polybft.SmartContract{
+			Address: types.Address(contract.expected),
+			Code:    input,
+			Name:    fmt.Sprintf("%s/%s", contract.chain, contract.name),
+		}
+
+		result = append(result, smartContract)
+	}
+
+	return result, nil
+}
+
+// GetInitialValidatorsDelta extracts initial account set from the genesis block and
+// populates validator set delta to its extra data
```

Comment:
The comment doesn't match what the method actually does - it doesn't populate the extra data of the genesis block, it just extracts the delta

Reply:
@igorcrevar Fixed where?